langwatch/src/components/evaluations/wizard/WizardWorkspace.tsx

```diff
@@ -170,6 +172,18 @@ export const WizardWorkspace = memo(function WizardWorkspace() {
                       datasetId={getDatasetId()}
                       insideWizard
                       gridRef={datasetGridRef}
+                      onUpdateDataset={(updatedDataset) => {
+                        // Update the wizard state when dataset is modified
+                        if (
+                          updatedDataset.datasetId &&
+                          updatedDataset.columnTypes
+                        ) {
+                          setDatasetId(
+                            updatedDataset.datasetId,
+                            updatedDataset.columnTypes
+                          );
+                        }
+                      }}
```

Comment:
Why can't this be managed in the wizardstore/wizardstore hooks directly, instead of lost in the TSX markup?